# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1295722458 18000
#      Sat Jan 22 13:54:18 2011 -0500
# Node ID adbc08052ccdd05f812b68191a33210d7d9db213
# Parent  dc11166fcddc330d98e2a4a1bed8fe7ed333bb6d
meshc.m: Add note describing why the gnuplot backend is not consistent with Matlab.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,10 +1,12 @@
 2011-01-22  Ben Abbott  <bpabbott@mac.com>
 
+	* plot/meshc.m: Add note: gnuplot does not support filled 3D patches,
+	or mixing non-filled contours with filled surfaces.
 	* plot/subplot.m: Set "box" to "on" by default (same as Matlab).
 	* plot/cylinder.m: Remove console output from demo.
 
 2011-01-22  Konstantinos Poulios  <logari81@googlemail.com>
 
 	* plot/plotyy.m: Set box property to off to allow both y-axes colors
 	to be visible for OpenGL backends.
 
diff --git a/scripts/plot/meshc.m b/scripts/plot/meshc.m
--- a/scripts/plot/meshc.m
+++ b/scripts/plot/meshc.m
@@ -32,16 +32,20 @@ function h = meshc (varargin)
   newplot ();
 
   tmp = surface (varargin{:});
 
   ax = get (tmp, "parent");
 
   set (tmp, "facecolor", "w");
   set (tmp, "edgecolor", "flat");
+  ## FIXME - gnuplot does not support a filled surface and a
+  ## non-filled contour. 3D filled patches are also not supported.
+  ## Thus, the facecolor will be transparent for the gnuplot
+  ## backend.
 
   if (! ishold ())
     set (ax, "view", [-37.5, 30],
          "xgrid", "on", "ygrid", "on", "zgrid", "on");
   endif
 
   drawnow ();
   zmin = get (ax, "zlim")(1);
